NEWS.md

```diff
@@ -1,6 +1,50 @@
 JuMP release notes
 ==================
 
+Version 0.21 (Feb 13, 2020)
+-----------------------------
+
+Breaking changes:
+
+- Deprecated `with_optimizer` (#2090, #2084, #2141). You can replace
+  `with_optimizer` by either nothing, `optimizer_with_attributes` or a closure:
+  * replace `with_optimizer(Ipopt.Optimizer)` by `Ipopt.Optimizer`.
+  * replace `with_optimizer(Ipopt.Optimizer, max_cpu_time=60.0)`
+    by `optimizer_with_attributes(Ipopt.Optimizer, "max_cpu_time" => 60.0)`.
+  * replace `with_optimizer(Gurobi.Optimizer, env)` by `() -> Gurobi.Optimizer(env)`.
+  * replace `with_optimizer(Gurobi.Optimizer, env, Presolve=0)`
+    by `optimizer_with_attributes(() -> Gurobi.Optimizer(env), "Presolve" => 0)`.
+
+  alternatively to `optimizer_with_attributes`, you can also set the attributes
+  separately with `set_optimizer_attribute`.
+- Renamed `set_parameter` to `set_optimizer_attribute` (#2150).
+- Broadcast should now be explicit inside macros. `SDconstraint(model, x >= 1)`
+  and `@constraint(model, x + 1 in SecondOrderCone())` now throw an error
+  instead of broadcasting `1` along the dimension of `x` (#2107).
+- `SDconstraint(model, x >= 0)` is now equivalent to `constraint(model, x in PSDCone())`
+  instead of `constraint(model, (x .- 0) in PSDCone())` (#2107).
+- The macros now create the containers with `map` instead of `for` loops,
+  as a consequence, containers created by `@expression` can now have any element
+  type and containers of constraint references now have concrete element types
+  when possible (#2070).
```

Comment:
I would add more: This fixes a long-standing issue where `@expression` could only be used to generate a collection of linear expressions. Now it works for quadratic expressions as well.